### PR TITLE
feat: [SSCA-608]: Added Image in DockerArtifactOutcome

### PR DESCRIPTION
**Pull request by arpit-harness**
## Describe your changes
Added Image( "DomainName/ImagePath:tag") in DockerArtifactOutcome, this is done so that is starts showing up in instanceNG collection from where SSCA-Module will read using a debezium service and reflect in artifacts store in SSCA DB.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53455)

### CODE REVIEW

**Comment by arpit-harness** [comment]
trigger uts

**Review by deepharness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53455)* status: 6 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`126-instance/src/main/java/io/harness/entities/ArtifactDetails.java` line 25 at r1](https://reviewable.io/reviews/harness/harness-core/53455#-NfP2oY223L7j3nbZ9NY:-NfP2oY223L7j3nbZ9NZ:b-6bpn54) ([raw file](https://github.com/harness/harness-core/blob/a9767448e1ba0c9d6bc4dab182b2702b27df2f5e/126-instance/src/main/java/io/harness/entities/ArtifactDetails.java#L25)):*
> ```Java
>   private String artifactId;
>   private String tag; // this corresponds to the build number of the artifact
>   private String image; // this corresponds to complete image path of the artifact
> ```

why don't we call it url ? as artifact will represent both image & repo in future

**Review by arpit-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53455)* status: 6 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`126-instance/src/main/java/io/harness/entities/ArtifactDetails.java` line 25 at r1](https://reviewable.io/reviews/harness/harness-core/53455#-NfP2oY223L7j3nbZ9NY:-NfPpcFp7cj4GUO9ykTc:b88215w) ([raw file](https://github.com/harness/harness-core/blob/a9767448e1ba0c9d6bc4dab182b2702b27df2f5e/126-instance/src/main/java/io/harness/entities/ArtifactDetails.java#L25)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

why don't we call it url ? as artifact will represent both image & repo in future
</blockquote></details>

"Image" is used in the CD code. Also this is not exactly a url.. its of format DomainName/ImagePath:Tag where DomainName is just the host part of the url.

File: 120-ng-manager/config.yml [hunk 1]
```diff
@@ -287,7 +287,7 @@ pmsGrpcClientConfig:
   target: localhost:12011
   authority: localhost
 
-shouldConfigureWithPMS: ${SHOULD_CONFIGURE_WITH_PMS:-false}
```
Comment (puthrayaharness) [inline 120-ng-manager/config.yml:290]: Please revert

File: 126-instance/src/main/java/io/harness/entities/ArtifactDetails.java [hunk 2]
```diff
@@ -22,4 +22,5 @@ public class ArtifactDetails {
   private String displayName;
   private String artifactId;
   private String tag; // this corresponds to the build number of the artifact
+  private Object artifactIdentity; // this corresponds to complete image path of the artifact
```
Comment (puthrayaharness) [inline 126-instance/src/main/java/io/harness/entities/ArtifactDetails.java:25]: When I mentioned it should be an Object, I meant to make a generic class and add fields. Not make it Object

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Review by puthrayaharness — DISMISSED** [review]

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger uts

**Review by inderschera — APPROVED** [review]

**Review by TarunUba1 — APPROVED** [review]

**Review by TarunUba1 — APPROVED** [review]

**Review by TarunUba1 — COMMENTED** [review]
add tests

**Pull request merged**